sysdata/csv/csv_instrument_data.py

```diff
@@ -91,7 +93,11 @@ def _add_instrument_data_without_checking_for_existing_entry(
         )
 
     def write_all_instrument_data_from_df(self, instrument_data_as_df: pd.DataFrame):
-        instrument_data_as_df.to_csv(self._config_file, index_label="Instrument")
+        instrument_data_as_df.to_csv(
+            self._config_file,
+            index_label="Instrument",
+            columns=[field.name for field in dataclasses.fields(instrumentMetaData)]
```

Comment:
we probably want an explicit check here for membership of instrument_data_as_df, e.g.
```
 columns=[field.name for field in dataclasses.fields(instrumentMetaData) if field.name in instrument_data_as_df.columns]
```